core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoRequestManager.java

```diff
@@ -1500,7 +1511,24 @@ public long saveProjectOverviewNote(ProjectOverview overview, Note note) throws
   @Override
   public long saveRun(Run run) throws IOException {
     if (runStore != null) {
-      return runStore.save(run);
+      long id;
+      if (run.getId() == AbstractRun.UNSAVED_ID) {
+        run.setId(runStore.save(run));
```

Comment:
This will actually fail due to null name. Should generate a temporary name before initial save, like in `DefaultSampleService.create(Sample)`